services/system/Accounts/plugin/smartAuth/export.wit

```diff
@@ -0,0 +1,53 @@
+package accounts:smart-auth;
+
+interface types {
+    /// A list of `claim` objects is added to every submitted transaction.
+    /// A claim is something the submitter of a transaction claims to know (e.g. a public key)
+    record claim {
+        /// The name of the service used to verify the claim is valid
+        verify-service: string,
+        /// The raw claim data
+        raw-data: list<u8>,
+    }
+
+    /// The verify-service specified in the claim is the service that knows how to verify that
+    /// a proof actually proves the claim.
+    record proof {
+        signature: list<u8>,
+    }
+
+    /// Action data from a transaction
+    record action {
+        /// The service that will execute this action
+        service: string,
+        /// The name of the action to execute
+        method: string, 
+        /// The data passed to the action, packed in fracpack format
+        raw-data: list<u8>,
+    }
+}
+
+/// Export this interface to implement an auth service plugin
+/// This interface should only be exported, not imported.
+interface smart-auth {
+    use host:common/types.{error};
+    use types.{claim, proof, action};
+
+    /// Gets the claim(s) needed to satisfy the authorization of the sender of this transaction
+    /// 
+    /// Parameters
+    /// * `account-name`: The name of the account whose authorization this transaction
+    ///   claims to satisfy (The sender of the transaction).
+    /// * `actions`: The actions that the sender of this transaction is attempting to execute
+    get-claims: func(account-name: string, actions: list<action>) -> result<list<claim>, error>;
+
+    /// Provides proof(s) of the claim(s). 
+    /// If there are multiple claims, the returned list of proofs must be in the same order as the 
+    /// claims were specified.
+    /// 
+    /// Parameters
+    /// * `account-name`: The name of the account whose claim is being proven
+    /// * `message`: Data needed to generate the proof. Often a message that must be 
```

Comment:
>  Don't the caller and callee need to interpret message in the same way?

I don't think that's necessarily true. The way this is implemented, that *is* what it will always be (unless they write their own `transact` service/plugin), because `transact` is involved.

Maybe this implies that the construction of the transaction should be entirely up to the auth service. What if the API was just, "auth_service::make_transaction(sender, actions)". Then the auth service adds claims, adds notifier claims, hashes, adds proofs, adds notifier proofs. 

Is that a more sensible abstraction? It keeps `transact` unopinionated on how proofs are generated.

Reply:
> The description differs from import.wit

Done

Reply:
I don't think that the auth service should be responsible for building the transaction. That would still leave the notifier and the auth services needing to agree.

Reply:
Okay, in that case... regarding:

> Does it actually make sense for message to be anything else? Don't the caller and callee need to interpret message in the same way?

I don't know. `message` will, for now, always be a hash of the transaction object, but it may be possible for the callee to ignore it during proof generation, as long as the provided proof is verifiable, right?

If true, then what about this wording?
```
  /// * `message`: A hash of the current transaction. Often signed to generated a proof.
```
I could rename message to `tx-hash`, even. Thoughts?

Reply:
For now, it only makes sense for it to be a hash. It should match the argument passed to the verify service, which is `transactionHash` and which is calculated by native.